orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -1311,3 +1311,41 @@ def forward(self, input1):
 
     assert output['out'] is not None
     assert output['none_output'] is None
+
+def test_model_partially_requires_grad():
```

Comment:
Can you add a test where the `requires_grad_` is called on one of the submodules after the first call to `ORTModule.forward()`?

Reply:
Do you mean when ```requires_grad_``` chacges between one ORTModule forward call to another?

Reply:
I've added support for the scenario where users may want to change the requires_grad attribute in the middle of training. And have also added a unit test for the same.
Let me know if this is not what you intended.